packages/cli/src/cmds/cmd_build/mod.rs

```diff
@@ -42,28 +42,68 @@ use generate::generate_cxx_bridge_src;
 // A rust crate that will be built as a component of the megaton lib or the mod
 #[allow(dead_code)]
 struct RustCrate {
-    path: PathBuf,
-    manifest: RustManifest,
+    manifest: PathBuf,
+    got_built: bool,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-struct RustManifest {
-    // TODO: Implement
-}
+impl RustCrate {
+    pub fn new(manifest_path: PathBuf) -> Self {
+        Self {
+            manifest: manifest_path.clone().canonicalize().expect(format!("Could not find Cargo.toml at {:?}", manifest_path).as_str()),
+            got_built: true,
+        }
+    }
 
-impl RustManifest {
-    fn load(_crate_path: &Path) -> Self {
-        // TODO: Implement
-        Self {}
+    pub fn build(&mut self, build: config::Build) -> cu::Result<()> {
+        let cargo = cu::which("cargo").context("cargo executable not found")?;
+        let rust_flags = match build.flags.rust {
+            Some(flags) => Some(flags.join(" ")),
+            None => None
+        };
+        let mut command = cargo.command()
+            .add(cu::args![
+                "+megaton",
+                "build",
+                "--manifest_path",
+                &self.manifest,
+            ])
+            .stdin_null()
```

Comment:
there's actually a preset to handle cargo build output with animation and everything https://cu.pistonite.dev/pistonite_cu/pio/fn.cargo.html